spirv_glsl.cpp

```diff
@@ -3584,6 +3581,15 @@ void CompilerGLSL::emit_extension_workarounds()
 			}
 			statement("#endif");
 		}
+		else if (shader_subgroup_supporter.is_feature_requested(Supp::SubgroupMemBarrier))
+		{
+			statement("#ifndef GL_KHR_shader_subgroup_basic");
```

Comment:
I'd issue more statements to obtain a 
```glsl
#ifdef GL_KHR_shader_subgroup_basic
#else
// the stuff
#endif
```

Reply:
Actually you'd need to discriminate if you're compiling a Compute / Tessellation Eval Shader or the rest.

Because if its a shader with workgroups (Compute/TES) then you can replace all the barriers by `groupMemoryBarrier` because 

> groupMemoryBarrier waits on the completion of all memory accesses performed by an invocation of a compute shader relative to the same access performed by other invocations in the same work group and then returns with no other effect.